pkg/agent/run.go

```diff
@@ -190,7 +204,44 @@ func Run(cmd *cobra.Command, args []string) {
 	}
 }
 
-func gatherAndOutputData(config CombinedConfig, preflightClient client.Client, dataGatherers map[string]datagatherer.DataGatherer) {
+// Creates an event recorder for the agent's Pod object. Expects the env var
+// POD_NAME to contain the pod name. Note that the RBAC rule allowing sending
+// events is attached to the pod's service account, not the impersonated service
+// account (venafi-connection).
+func newEventf(installNS string) (Eventf, error) {
+	restcfg, err := kubeconfig.LoadRESTConfig("")
+	if err != nil {
+		logs.Log.Fatalf("failed to load kubeconfig: %v", err)
+	}
+	scheme := runtime.NewScheme()
+	_ = corev1.AddToScheme(scheme)
+
+	var eventf Eventf
+	if os.Getenv("POD_NAME") == "" {
+		eventf = func(eventType, reason, msg string, args ...interface{}) {}
+		logs.Log.Printf("error messages will not show in the pod's events because the POD_NAME environment variable is empty")
+	} else {
+		podName := os.Getenv("POD_NAME")
+
+		eventClient, err := kubernetes.NewForConfig(restcfg)
+		if err != nil {
+			return eventf, fmt.Errorf("failed to create event client: %v", err)
+		}
+		broadcaster := record.NewBroadcaster()
+		broadcaster.StartRecordingToSink(&clientgocorev1.EventSinkImpl{Interface: eventClient.CoreV1().Events(installNS)})
+		eventRec := broadcaster.NewRecorder(scheme, corev1.EventSource{})
+		eventf = func(eventType, reason, msg string, args ...interface{}) {
+			eventRec.Eventf(&corev1.Pod{ObjectMeta: v1.ObjectMeta{Name: podName, Namespace: installNS}}, eventType, reason, msg, args...)
```

Comment:
I expected to see the events included in the output of `kubectl describe pod -l app.kubernetes.io/name=venafi-kubernetes-agent` but they were missing. 
Perhaps because this metadata does not include the POD UID?

I applied this patch and it allowed me to see the events with kubectl describe pod

```diff
diff --git a/deploy/charts/venafi-kubernetes-agent/templates/deployment.yaml b/deploy/charts/venafi-kubernetes-agent/templates/deployment.yaml
index cb780e7..0115c1a 100644
--- a/deploy/charts/venafi-kubernetes-agent/templates/deployment.yaml
+++ b/deploy/charts/venafi-kubernetes-agent/templates/deployment.yaml
@@ -98,6 +98,10 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.name
+          - name: POD_UID
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.uid
           {{- if .Values.metrics.enabled }}
           ports:
             - containerPort: 8081
diff --git a/pkg/agent/run.go b/pkg/agent/run.go
index 8877c5e..0fa8038 100644
--- a/pkg/agent/run.go
+++ b/pkg/agent/run.go
@@ -21,6 +21,7 @@ import (
        corev1 "k8s.io/api/core/v1"
        v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
        "k8s.io/apimachinery/pkg/runtime"
+       "k8s.io/apimachinery/pkg/types"
        "k8s.io/client-go/kubernetes"
        clientgocorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
        "k8s.io/client-go/tools/record"
@@ -231,7 +232,7 @@ func newEventf(installNS string) (Eventf, error) {
                broadcaster.StartRecordingToSink(&clientgocorev1.EventSinkImpl{Interface: eventClient.CoreV1().Events(installNS)})
                eventRec := broadcaster.NewRecorder(scheme, corev1.EventSource{})
                eventf = func(eventType, reason, msg string, args ...interface{}) {
-                       eventRec.Eventf(&corev1.Pod{ObjectMeta: v1.ObjectMeta{Name: podName, Namespace: installNS}}, eventType, reason, msg, args...)
+                       eventRec.Eventf(&corev1.Pod{ObjectMeta: v1.ObjectMeta{Name: podName, Namespace: installNS, UID: types.UID(os.Getenv("POD_UID"))}}, eventType, reason, msg, args...)
                }
        }


```

Reply:
Not sure. What did you find when you tested it?